exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetComparisonPredicate.java

```diff
@@ -83,23 +84,26 @@ private ParquetComparisonPredicate(
    * where Column1 and Column2 are from same parquet table.
    */
   @Override
-  public boolean canDrop(RangeExprEvaluator<C> evaluator) {
+  public RowsMatch matches(RangeExprEvaluator<C> evaluator) {
     Statistics<C> leftStat = left.accept(evaluator, null);
-    if (isNullOrEmpty(leftStat)) {
-      return false;
+    if (isNullOrEmpty(leftStat) || !leftStat.hasNonNullValue()) {
```

Comment:
The check for `hasNonNullValue()` should be done after check for `isAllNulls()` or, otherwise, it will always return `SOME` when all values are null.

Reply:
Done